src/sql/workbench/contrib/notebook/test/browser/notebookLinkHandler.test.ts

```diff
@@ -103,4 +103,15 @@ suite('Noteboook Link Handler', function (): void {
 		result = new NotebookLinkHandler(notebookUri, Object.assign(document.createElement('a'), { href: '/tmp/my%20stuff.png' }), configurationService);
 		assert.strictEqual(result.getLinkUrl(), `.${path.sep}my%2520stuff.png`, 'Basic link test with %20 filename failed');
 	});
+
+	test('Should return correctly encoded url/filePath', () => {
+		let result = new NotebookLinkHandler(notebookUri, 'https://github.com/search/advanced?q=test&r=microsoft%2Fazuredatastudio&type=Code', configurationService);
+		assert.strictEqual(result.getEncodedLinkUrl(result.getLinkUrl()), `https://github.com/search/advanced?q=test&r=microsoft%2Fazuredatastudio&type=Code`, 'HTTPS link encoding failed');
+
+		result = new NotebookLinkHandler(notebookUri, 'https://github.com/search/advanced?q=test&r=(microsoft%2Fazuredatastudio)&type=Code', configurationService);
+		assert.strictEqual(result.getEncodedLinkUrl(result.getLinkUrl()), `https://github.com/search/advanced?q=test&r=(microsoft%2Fazuredatastudio)&type=Code`, '() in HTTP link failed to resolve');
+
+		result = new NotebookLinkHandler(notebookUri, 'https://github.com/search/advanced?q=test&r=(microsoft/azuredata studio)&type=Code', configurationService);
+		assert.strictEqual(result.getEncodedLinkUrl(result.getLinkUrl()), `https://github.com/search/advanced?q=test&r=(microsoft/azuredata%20studio)&type=Code`, 'space failed to be encoded failed');
```

Comment:
Add tests that test passing in an already-encoded URI

Reply:
the first test is that, passing in the already encoded '/' as %2F in the URL and checking that it's not re-encoding the % in it.